apps/api/src/modules/notifications/notifications.service.ts

```diff
@@ -109,8 +114,18 @@ export class NotificationsService extends CoreService<Notification> {
 
     for (const notification of allPendingNotifications) {
       try {
-        notification.deliveryStatus = DeliveryStatus.IN_PROGRESS;
-        await this.notificationQueueService.addNotificationToQueue(notification);
+        if (notification.retryCount <= this.maxRetryCount) {
```

Comment:
No need to add this check here. Its better to do it at the place I mentioned so that we don't fetch much records from notifications and do check for all records. We make sure all PENDING ones are valid for RETRY

Reply:
This is addressed in https://github.com/OsmosysSoftware/osmo-x/pull/263/commits/1c8fee46e6cd7bb826b8d193fbd6d1f580a3f9c7